<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CMvRingDefect</class>
 <widget class="QWidget" name="CMvRingDefect">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>892</width>
    <height>652</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>圆环缺口</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_3">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <widget class="QGraphicsView" name="graphicsView_mainImage">
       <property name="minimumSize">
        <size>
         <width>500</width>
         <height>600</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="frame">
       <property name="minimumSize">
        <size>
         <width>500</width>
         <height>42</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>500</width>
         <height>42</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>2</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QPushButton" name="pushButton_zoomIn">
          <property name="minimumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="text">
           <string>放大</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_zoomOut">
          <property name="minimumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="text">
           <string>缩小</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_zoomBest">
          <property name="minimumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="text">
           <string>最好尺寸</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_lockRoiMove">
          <property name="minimumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>70</width>
            <height>40</height>
           </size>
          </property>
          <property name="text">
           <string>锁定ROI</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>63</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="label_mousePos">
          <property name="minimumSize">
           <size>
            <width>150</width>
            <height>40</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>150</width>
            <height>40</height>
           </size>
          </property>
          <property name="text">
           <string>TextLabel</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="minimumSize">
        <size>
         <width>382</width>
         <height>602</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tab">
        <attribute name="title">
         <string>输入设置</string>
        </attribute>
        <widget class="QLabel" name="label_funcName">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>30</y>
           <width>81</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>检测器名称：</string>
         </property>
        </widget>
        <widget class="QCheckBox" name="checkBox_enableFunc">
         <property name="geometry">
          <rect>
           <x>270</x>
           <y>20</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>101</width>
           <height>31</height>
          </size>
         </property>
         <property name="text">
          <string>启用检测器</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QPlainTextEdit" name="plainTextEdit_funcName">
         <property name="geometry">
          <rect>
           <x>130</x>
           <y>20</y>
           <width>131</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>131</width>
           <height>31</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>131</width>
           <height>31</height>
          </size>
         </property>
         <property name="overwriteMode">
          <bool>false</bool>
         </property>
        </widget>
        <widget class="QTableWidget" name="tableWidget_input">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>70</y>
           <width>341</width>
           <height>391</height>
          </rect>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="textElideMode">
          <enum>Qt::ElideRight</enum>
         </property>
         <property name="verticalScrollMode">
          <enum>QAbstractItemView::ScrollPerItem</enum>
         </property>
         <property name="showGrid">
          <bool>false</bool>
         </property>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderCascadingSectionResizes">
          <bool>false</bool>
         </attribute>
         <row>
          <property name="text">
           <string>1</string>
          </property>
         </row>
         <column>
          <property name="text">
           <string>输入名称</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>输入来源</string>
          </property>
         </column>
         <item row="0" column="0">
          <property name="text">
           <string>黑白图像</string>
          </property>
         </item>
         <item row="0" column="1">
          <property name="text">
           <string>某个检测器的输出</string>
          </property>
         </item>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_2">
        <attribute name="title">
         <string>参数设置</string>
        </attribute>
        <widget class="QComboBox" name="comboBox_binaryDetection">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>50</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <item>
          <property name="text">
           <string>手动白色</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>手动黑色</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>自动白色</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>自动黑色</string>
          </property>
         </item>
        </widget>
        <widget class="QLabel" name="label_binaryDetection">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>111</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>二值化类型：</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_grayLevel">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>130</x>
           <y>20</y>
           <width>111</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>灰度阈值：</string>
         </property>
        </widget>
        <widget class="QSpinBox" name="spinBox_grayLevel">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>130</x>
           <y>50</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="maximum">
          <number>255</number>
         </property>
        </widget>
        <widget class="QGroupBox" name="groupBox_curvature">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>310</y>
           <width>241</width>
           <height>81</height>
          </rect>
         </property>
         <property name="title">
          <string>曲率筛选</string>
         </property>
         <widget class="QLabel" name="label_curvatureMin">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>10</y>
            <width>111</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>曲率下限：</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_curvatureMax">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>10</y>
            <width>111</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>曲率上限：</string>
          </property>
         </widget>
         <widget class="QSpinBox" name="spinBox_curvatureMax">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>40</y>
            <width>101</width>
            <height>31</height>
           </rect>
          </property>
          <property name="minimum">
           <number>-360</number>
          </property>
          <property name="maximum">
           <number>360</number>
          </property>
         </widget>
         <widget class="QSpinBox" name="spinBox_curvatureMin">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>40</y>
            <width>101</width>
            <height>31</height>
           </rect>
          </property>
          <property name="minimum">
           <number>-360</number>
          </property>
          <property name="maximum">
           <number>360</number>
          </property>
         </widget>
        </widget>
        <widget class="QGroupBox" name="groupBox_area">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>210</y>
           <width>361</width>
           <height>81</height>
          </rect>
         </property>
         <property name="title">
          <string>面积筛选</string>
         </property>
         <widget class="QLabel" name="label_areaMax">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>10</y>
            <width>111</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>面积上限：</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_areaMin">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>10</y>
            <width>111</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>面积下限：</string>
          </property>
         </widget>
         <widget class="QSpinBox" name="spinBox_areaMin">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>40</y>
            <width>101</width>
            <height>31</height>
           </rect>
          </property>
          <property name="maximum">
           <number>5000000</number>
          </property>
         </widget>
         <widget class="QSpinBox" name="spinBox_areaMax">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>40</y>
            <width>101</width>
            <height>31</height>
           </rect>
          </property>
          <property name="maximum">
           <number>5000000</number>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButton_areaInformation">
          <property name="geometry">
           <rect>
            <x>250</x>
            <y>40</y>
            <width>101</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>显示面积信息</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_initialize">
          <property name="geometry">
           <rect>
            <x>250</x>
            <y>10</y>
            <width>111</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>面积信息：</string>
          </property>
         </widget>
        </widget>
        <widget class="QLabel" name="label_titleJudged">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>140</y>
           <width>381</width>
           <height>31</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:0, stop:0 rgba(80, 80, 80, 255), stop:1 rgba(120, 120, 120, 120));
color: rgb(255, 255, 255);</string>
         </property>
         <property name="text">
          <string>  结果判定</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
        <widget class="QGroupBox" name="groupBox_dimension">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>410</y>
           <width>241</width>
           <height>81</height>
          </rect>
         </property>
         <property name="title">
          <string>尺度筛选</string>
         </property>
         <widget class="QLabel" name="label_dimensionMin">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>10</y>
            <width>111</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>尺度下限：</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_dimensionMax">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>10</y>
            <width>111</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>尺度上限：</string>
          </property>
         </widget>
         <widget class="QSpinBox" name="spinBox_dimensionMax">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>40</y>
            <width>101</width>
            <height>31</height>
           </rect>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
         </widget>
         <widget class="QSpinBox" name="spinBox_dimensionMin">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>40</y>
            <width>101</width>
            <height>31</height>
           </rect>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
         </widget>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_4">
        <attribute name="title">
         <string>结果绘制</string>
        </attribute>
        <widget class="QCheckBox" name="checkBox_valuTrig">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>20</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>启用绘制</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </widget>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="leftMargin">
        <number>1</number>
       </property>
       <property name="topMargin">
        <number>1</number>
       </property>
       <property name="rightMargin">
        <number>1</number>
       </property>
       <property name="bottomMargin">
        <number>1</number>
       </property>
       <item>
        <widget class="QPushButton" name="pushButton_ones">
         <property name="minimumSize">
          <size>
           <width>80</width>
           <height>40</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>单次</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_affirm">
         <property name="minimumSize">
          <size>
           <width>80</width>
           <height>40</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>确定</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pbPrev_cancel">
         <property name="minimumSize">
          <size>
           <width>80</width>
           <height>40</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>取消</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="CMvContourMatching.qrc"/>
 </resources>
 <connections/>
</ui>
